packages/odk/src/Adaptor.js

```diff
@@ -0,0 +1,148 @@
+import { expandReferences } from '@openfn/language-common/util';
+import { execute as commonExecute } from '@openfn/language-common';
+import * as util from './Utils';
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute`, and prepends initial state for commcare.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(
+      util.authorize,
+      ...operations
+    )({ ...initialState, ...state });
+  };
+}
+
+/**
+ * Make a GET request
+ * @example
+ * get("v1/projects");
+ * @function
+ * @param {string} path - Path to resource
+ * @param {Object} params - Optional request params
+ * @param {Object} headers - Optional request headers
+ * @param {function} [callback] - Optional callback to handle the response
+ * @returns {Operation}
+ */
+export function get(path, params, headers, callback) {
+  return request('GET', path, null, params, callback, headers);
+}
+
+/**
+ * Make a GET request
+ * @example
+ * getForms(22);
+ * @function
+ * @param {number} id - Id of the project you want to get its forms
+ * @param {Object} params - Optional request params
+ * @param {Object} headers - Optional request headers
+ * @param {function} [callback] - Optional callback to handle the response
+ * @returns {Operation}
+ */
+export function getForms(id, params, headers, callback) {
+  const path = `/v1/projects/${id}/forms`;
+  return request('GET', path, null, params, callback, headers);
+}
+
+/**
+ * Make a GET request
+ * @example
+ * getSubmissions(22, 'test');
+ * @function
+ * @param {number} projectId - Id of the project you want to get its forms submissions
+ * @param {string} xmlFormId - Id of the form you want to get its submissions
+ * @param {Object} params - Optional request params
+ * @param {Object} headers - Optional request headers
+ * @param {function} [callback] - Optional callback to handle the response
+ * @returns {Operation}
+ */
+export function getSubmissions(
+  projectId,
+  xmlFormId,
+  params,
```

Comment:
I don't think we should support params and headers here. The whole idea if this function is to abstract out the http layer. The function should add whatever params and headers are needed

Reply:
This is also true on `getForms`